nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -563,47 +568,115 @@ type CollectionStepFn = Box<
         + Send,
 >;
 
-enum CollectionStepOutput {
-    Ereports(SupportBundleEreportStatus),
-    SavingSpDumps { listed_sps: bool },
-    // NOTE: The distinction between this and "Spawn" is pretty artificial -
-    // it's just to preserve a part of the report which says "we tried to
-    // list in-service sleds".
-    //
-    // If we changed the collection report, this could easily be combined
-    // with the "Spawn" variant.
-    SpawnSleds { extra_steps: Vec<(&'static str, CollectionStepFn)> },
-    Spawn { extra_steps: Vec<(&'static str, CollectionStepFn)> },
-    None,
+struct CollectionStep {
+    name: String,
+    step_fn: CollectionStepFn,
+}
+
+impl CollectionStep {
+    fn new(name: impl Into<String>, step_fn: CollectionStepFn) -> Self {
+        Self { name: name.into(), step_fn }
+    }
+
+    async fn run(
+        self,
+        collection: &Arc<BundleCollection>,
+        output: &Utf8Path,
+    ) -> CompletedCollectionStep {
+        let start = Utc::now();
+
+        let output = (self.step_fn)(collection, output)
+            .await
+            .inspect_err(|err| {
+                warn!(
+                    collection.log,
+                    "Step failed";
+                    "name" => &self.name,
```

Comment:
nit, take it or leave it: perhaps this ought to be 
```suggestion
                    "step" => &self.name,
```

Reply:
Updated